SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialActorResolutionTest/SpatialActorResolutionTest.cpp

```diff
@@ -0,0 +1,58 @@
+﻿#include "SpatialActorResolutionTest.h"
+
+#include "Engine/NetDriver.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+
+ASelfDestroyingActor::ASelfDestroyingActor()
+{
+	bReplicates = true;
+}
+
+bool ASelfDestroyingActor::ReplicateSubobjects(UActorChannel* Channel, FOutBunch* Bunch, FReplicationFlags* RepFlags)
+{
+	TSharedPtr<FDelegateHandle> DelegateHandle = MakeShared<FDelegateHandle>();
+
+	// OnEndFrame decouples us from tick rates and gives no opportunity for the NetDriver to tick in-between.
+	*DelegateHandle = FCoreDelegates::OnEndFrame.AddLambda([this, DelegateHandle]() {
```

Comment:
IIRC this kind of  global delegate can tick for all worlds, so given a multiworker PIE session, it could tick several times? 
I think it's only an issue if you use things like GWorld, but that's something to keep in mind, only execute the delegate if GWorld == Actor->GetWorld()

Reply:
This might be problematic as it's executed outside of engine tick, so GWorld may well be non-initialized. I'll check this but it shouldn't be a problem either way.

Reply:
Yeah that is the case - it can have an arbitrary world in there, looks like the last world the engine ticked which is often the last client - never passes the condition and explodes once the test is failed. Adding protection against crashes/memory stomping in this case.